### PR TITLE
fix issue #9 

### PR DESCRIPTION
**Pull request by achangofpace**
- add new protocol `BACKGROUND_RESTORE_DEFAULT_MAPPINGS` to restore mappings in DB to defaults
- move some of the MappingSelectView constructor code into functions for reuse and readability
- change annotation content script to use a function that returns a template string as the second parameter to string.replace rather than the template string itself
- hide WIP "Show JSON" button
- update Contribution Guide with more instructions
- update Todo

### CODE REVIEW

File: database.js
Comment (achangofpace) [inline database.js:1]: formatting changes only

File: database.js [hunk 1]
```diff
@@ -563,7 +563,7 @@ const HANGGUL_TO_ROMAJA = {
 
 /* eslint-enable */
 
-/**
+/***
```
Comment (israbbani) [inline database.js:566]: Looks like a typo?

File: popup/choose_mapping.js [hunk 2]
```diff
@@ -632,19 +699,16 @@ class MappingsSelectionView {
...
-				);
+				return browser.runtime.sendMessage({
+					intended_recipient: RECIPIENT_BACKGROUND,
+					command: BACKGROUND_SAVE_MAPPINGS,
+					MAPPINGS_TO_SAVE: BACKGROUND_APPLY_MAPPINGS_PRIORITY_UPDATE_response.updated_mappings
+				});
 			}
 		})
 		.catch((err) => {
-			reject(err);
```
Comment (achangofpace) [inline popup/choose_mapping.js:647]: there was no one watching this promise to deal with rejection

File: popup/choose_mapping.js [hunk 3]
```diff
@@ -632,19 +699,16 @@ class MappingsSelectionView {
...
+			throw err;
```
Comment (achangofpace) [inline popup/choose_mapping.js:711]: failure to saveMappingsPriority should probably result in a user notification with an offer to try again.

File: package.json
Comment (achangofpace) [inline package.json:1]: whitespace only

File: content_scripts/content_annotation_handler.js [hunk 4]
```diff
@@ -90,7 +90,9 @@ function rubyWrap(node, regexs, consolidated_mapping) {
 					};
 					content = content.replace(
 						regex,
-						`<ruby data-ie-annotated='${JSON.stringify(annotation_tag)}' style="font-size: 1.5em;">${key}<rp>(</rp><rt style="font-size: 1em">${annotation}&nbsp;</rt><rp>)</rp></ruby>`
+						(match) => {
```
Comment (israbbani) [inline content_scripts/content_annotation_handler.js:93]: Can you rename the parameter something that indicates it's not used? People use `_` sometimes to refer to unused parameters.

File: popup/choose_mapping.js [hunk 5]
```diff
@@ -583,6 +533,89 @@ class MappingsSelectionView {
...
+					});
+				}
+			}
+			return Promise.all([
+				Promise.resolve(BACKGROUND_APPLY_MAPPINGS_PRIORITY_UPDATE_response.updated_mappings),
+				tabs
+			]);
+		})
+		.then(([mappings, tabs]) => {
+			return _annotate(tabs, mappings);
```
Comment (israbbani) [inline popup/choose_mapping.js:572]: `([mapping, tabs]) => _annotate(tabs, mappings)` is also legal

**Review by israbbani — APPROVED** [review]
LGTM. Minor comments if you want to change some stuff.

**Pull request merged**